keras/applications/regnet.py

```diff
@@ -0,0 +1,1349 @@
+    else:
+      conv_3x3 = layers.Conv2D(inv_btlneck_filters, (3, 3), use_bias=False,
+                              groups=groups, name=name + "_conv_3x3")
+
+    # Build block
+    # conv_1x1_1
+    x = layers.Conv2D(inv_btlneck_filters, (1, 1),
+                      use_bias=False, name=name + "_conv_1x1_1")(inputs)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_1x1_1_bn")(x)
+    x = tf.nn.silu(x)
+
+    # # conv_3x3
+    x = conv_3x3(x)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_3x3_bn")(x)
+    x = tf.nn.silu(x)
+
+    # Squeeze-Excitation block (https://arxiv.org/abs/1709.01507)
+    x = layers.GlobalAveragePooling2D(name=name + "_se_gap")(x)
+    x = layers.Reshape((1, 1, inv_btlneck_filters), name=name + "_se_reshape")(x)
+    x = layers.Conv2D(se_filters, (1, 1), activation=tf.nn.silu,
+                      name=name + "_se_squeeze")(x)
+    x = layers.Conv2D(inv_btlneck_filters, (1, 1),
+                      activation="sigmoid", name=name + "_se_excite")(x)
+
+    # conv_1x1_2
+    x = layers.Conv2D(filters_out, (1, 1), use_bias=False, 
+                      name=name + "_conv_1x1_2")(x)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_1x1_2_bn")(x)
+
+    if stride != 1:
+      return x
+    else:
+      return x + inputs
+  return apply
+
+
+def Stage(block_type,
+          depth,
+          group_width, 
+          filters_in, 
+          filters_out, 
+          name=""):
+  """Implementation of Stage in RegNet.
+
+  Args:
+    inputs: input tensor
+    block_type: must be one of "X", "Y", "Z"
+    depth: depth of stage, number of blocks to use
+    group_width: group width of all blocks in  this stage
+    filters_in: input filters to this stage
+    filters_out: output filters from this stage
+    name: name prefix
+
+  Returns:
+    Output tensor of Stage
+  """
+  def apply(inputs):
+    x = inputs
+    if block_type == "X":
+      x = XBlock(filters_in, filters_out, group_width, stride=2, name=name + "_XBlock_0")(x)
+      for i in range(depth - 1):
+        x = XBlock(filters_out, filters_out,
+                   group_width, name=name + "_XBlock_" + str(i))(x)
+    elif block_type == "Y":
+      x = YBlock(filters_in, filters_out, group_width,
+                 stride=2, name=name + "_YBlock_0")(x)
+      for i in range(depth - 1):
+        x = YBlock(filters_out, filters_out, group_width,
+                   name=name + "_YBlock_" + str(i))(x)
+    elif block_type == "Z":
+      x = ZBlock(filters_in, filters_out, group_width,
+                 stride=2, name=name + "_ZBlock_0")(x)
+      for i in range(depth - 1):
+        x = ZBlock(filters_out, filters_out, group_width,
+                   name=name + "_ZBlock_" + str(i))(x)
+    else:
+      raise NotImplementedError(f"""Block type {block_type} not implemented. 
```

Comment:
"not implemented" may be misleading, prefer saying "Block type not recognized". It could be a typo or a nonsensical string